src/onboarding2/Create/CreateStepsPanel.js

```diff
@@ -1,11 +1,43 @@
-import React from 'react'
+import React, { useMemo } from 'react'
 import PropTypes from 'prop-types'
 import { GU, useTheme } from '@aragon/ui'
 import CreateStepsItem from './CreateStepsItem'
 import CircleGraph from '../CircleGraph'
 
 function CreateStepsPanel({ step, steps }) {
   const theme = useTheme()
+
+  // Mark identical siblings to only show the last step
+  const [groupedSteps, displayedSteps] = useMemo(() => {
+    // these get updated by the .map() to avoid another iteration
+    let displayCount = 0
+
+    const groupedSteps = steps.map((step, index) => {
```

Comment:
This `map()` is really hard to reason about because of the reliance on mutations to an external `displayCount`. I'm also not really convinced we need to keep a 1:1 mapping between the given `steps` and `groupedSteps`, since we just filter out duplicates in `groupedSteps` (by not showing them).

What do you think about building this differently, and perhaps having a structure like

```js
step = {
  firstIndex: 0,
  lastIndex: 1, // can be the same as firstIndex, if the step's unique
}
```

Doing this via a `reduce()`, so we can build `groupedSteps` iteratively:

```js
const groupedSteps = steps.reduce((groupedSteps, step, index) => {
  const lastGroupedStepIndex = groupedSteps.length - 1
  if (index && groupedSteps[lastGroupedStepIndex] === step) {
    ++groupedSteps[lastGroupedStepIndex].lastIndex
  } else {
    groupedSteps.push({
      firstIndex: step,
      lastIndex: step,
    })
  }
  return groupedSteps
}, [])
```